test/integration/targets/cloudwatchlogs/tasks/main.yml

```diff
@@ -0,0 +1,138 @@
+        metric_transformation:
+          metric_name: box_free_space
+          metric_namespace: fluentd_metrics
+          metric_value: "$.value"
+      check_mode: yes
+      register: out
+
+    - name: check_mode state must be changed
+      assert:
+        that:
+          - out is changed
+
+    - name: set metric filter on '{{ log_group_name }}'
+      cloudwatchlogs_log_group_metric_filter:
+        log_group_name: '{{ log_group_name }}'
+        filter_name: '{{ filter_name }}'
+        filter_pattern: '{ ($.value = *) && ($.hostname = "box")}'
+        state: present
+        metric_transformation:
+          metric_name: box_free_space
+          metric_namespace: fluentd_metrics
+          metric_value: "$.value"
+      register: out
+
+    - name: create metric filter
+      assert:
+        that:
+          - out is changed
+
+    - name: re-set metric filter on '{{ log_group_name }}'
+      cloudwatchlogs_log_group_metric_filter:
+        log_group_name: '{{ log_group_name }}'
+        filter_name: '{{ filter_name }}'
+        filter_pattern: '{ ($.value = *) && ($.hostname = "box")}'
+        state: present
+        metric_transformation:
+          metric_name: box_free_space
+          metric_namespace: fluentd_metrics
+          metric_value: "$.value"
+      register: out
+
+    - name: metric filter must not change
+      assert:
+        that:
+          - out is not changed
+
+    - name: update metric transformation on '{{ log_group_name }}'
+      cloudwatchlogs_log_group_metric_filter:
+        log_group_name: '{{ log_group_name }}'
+        filter_name: '{{ filter_name }}'
+        filter_pattern: '{ ($.value = *) && ($.hostname = "box")}'
+        state: present
+        metric_transformation:
+          metric_name: box_free_space
+          metric_namespace: made_with_ansible
+          metric_value: "$.value"
+      register: out
+
+    - name: update metric filter
+      assert:
+        that:
+          - out is changed
+
+    - name: update filter_pattern on '{{ log_group_name }}'
+      cloudwatchlogs_log_group_metric_filter:
+        log_group_name: '{{ log_group_name }}'
+        filter_name: '{{ filter_name }}'
+        filter_pattern: '{ ($.value = *) && ($.hostname = "ansible")}'
+        state: present
+        metric_transformation:
+          metric_name: box_free_space
+          metric_namespace: made_with_ansible
+          metric_value: "$.value"
+      register: out
+
+    - name: update metric filter
+      assert:
+        that:
+          - out is changed
+
+    - name: checkmode delete metric filter on '{{ log_group_name }}'
+      cloudwatchlogs_log_group_metric_filter:
+        log_group_name: '{{ log_group_name }}'
+        filter_name: '{{ filter_name }}'
+        state: absent
+      check_mode: yes
+      register: out
+
+    - name: check_mode state must be changed
+      assert:
+        that:
+          - out is changed
+
+    - name: delete metric filter on '{{ log_group_name }}'
```

Comment:
It's also worth testing the behaviour of re-deleting an already absent filter  (should return "not changed")